client/landing/stepper/declarative-flow/site-setup-flow.ts

```diff
@@ -103,24 +104,137 @@ export const siteSetupFlow: Flow = {
 		const verticalsStepEnabled = isEnabled( 'signup/site-vertical-step' ) && isEnabledFTM;
 		const goalsStepEnabled = isEnabled( 'signup/goals-step' ) && isEnabledFTM;
 
-		// Set up Step progress for Woo flow - "Step 2 of 4"
-		if ( intent === 'sell' && storeType === 'power' ) {
+		updateFlowProgress();
+
+		function updateFlowProgress() {
```

Comment:
Is it possible to refactor this logic to be outside this function? There's a lot happening here 😅

Reply:
> Is it possible to refactor this logic to be outside this function? There's a lot happening here 😅

The `updateFlowProgress` should 

> For the currentStep, we can resolve the current stage (begin, middle, end) inside this function as you also define beginningSteps here.

I think it is similar to other method such as`submit`

Reply:
@taipeicoder I made it to a[ hook](https://github.com/Automattic/wp-calypso/blob/39d12c1511b8af9fdf02c1d4c2dcd51a08445ea2/client/landing/stepper/hooks/use-update-flow-progress.ts#L11)

Reply:
Nice! I've left several notes 🙂

Reply:
> 
> Personally, I think the progress bar looks odd by itself, since it contextually should be coupled with the navigational header 😅

I see your point now, but do you happen to know why some of the header is sticky and some not?
Also the [isSticky prop](https://github.com/Automattic/wp-calypso/blob/1acd2ad490b8f540784d029b4fc53e9e84663a07/client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/site-setup-design-picker.tsx#L138) for the header is managed inside step container while the progress bar is part of the flowRenderer, I wonder if we need to use Redux store to know wether the progress bar should be fixed position as well... I'm thinking about it

Reply:
I see what you mean, we can leave the progress bar non-sticky for now and follow-up in a future PR if needed (so far it's just me mentioning it 🙂)